### PR TITLE
update LeafObjects when list is completed and tracking finished lists

### PR DESCRIPTION
**Pull request by abhishek2021005**

### CODE REVIEW

File: packages/alea-frontend/pages/guided-tour2/[id].tsx [hunk 1]
```diff
@@ -240,13 +252,39 @@ async function stateTransition(
...
+      type: 'self-assessment',
+      concept: currentConceptUri,
+      competences: { Remember: 1.0, Understand: 1.0, Apply: 1.0 },
+      time: new Date().toISOString(),
+      payload: '',
+      comment: 'Self assessment through guided tour',
+    };
+    await updateLearnerModel(updatePayload);
+    if (
+      newState.completedConceptUris[newState.completedConceptUris.length - 1] ===
```
Comment (abchugh) [inline packages/alea-frontend/pages/guided-tour2/[id].tsx:278]: Why not just check `if (currentConceptUri ===  newState.targetConceptUri)`

File: packages/api/src/lib/lms.ts [hunk 2]
```diff
@@ -497,3 +497,18 @@ export async function getLeafConcepts(target: string) {
     target,
   })) as GetLeafConceptsResponse;
 }
+
+export interface LearnerModelUpdateEvent extends LMS2Event {
```
Comment (abchugh) [inline packages/api/src/lib/lms.ts:501]: We already have `SelfAssessmentEvent`

Comment (abhishek2021005) [reply packages/api/src/lib/lms.ts]: modified.

File: packages/alea-frontend/pages/guided-tour2/[id].tsx (hunk 1 above)
Comment (abhishek2021005) [reply packages/alea-frontend/pages/guided-tour2/[id].tsx]: implemented

**Review by abchugh — APPROVED** [review]

**Pull request merged**